### PR TITLE
get topic handler by communication message type

### PR DESCRIPTION
**Pull request by osiievska23**

### CODE REVIEW

File: src/main/java/com/icthh/xm/tmf/ms/communication/messaging/template/AbstractMessageTemplateService.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+    private Map<String, Object> getMessageModel(CommunicationMessageCreate message) {
+        return message.getCharacteristic().stream()
+            .filter(c -> TEMPLATE_MODEL_MESSAGE_CHARACTERISTIC.equals(c.getName()))
+            .findFirst()
+            .map(c -> getTemaplteModelMap(c.getValue()))
+            .orElse(Map.of());
+    }
+
+    private Map<String, Object> getTemaplteModelMap(String config) {
```
Comment (sergeysenja1992) [inline src/main/java/com/icthh/xm/tmf/ms/communication/messaging/template/AbstractMessageTemplateService.java:99]: очепятка

File: src/test/java/com/icthh/xm/tmf/ms/communication/messaging/template/TwilioMessageTemplateServiceUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        });
+
+        assertThrows("Language key, template name and tenant must be not blank",
+            IllegalStateException.class, () -> {
+                twilioMessageTemplateService.getMessageContent(TENANT, null, LOCALE, model);
+            });
+    }
+
+    @Test
+    public void getMessageContent_invalidTemplate_shouldThrowException() throws JsonProcessingException {
```
Comment (sergeysenja1992) [inline src/test/java/com/icthh/xm/tmf/ms/communication/messaging/template/TwilioMessageTemplateServiceUnitTest.java:80]: body of test and test name does not much

File: src/test/java/com/icthh/xm/tmf/ms/communication/service/TwilioServiceTest.java [hunk 3]
```diff
@@ -61,4 +64,32 @@ void send() {
         assertThat(test.getStatus()).isNotBlank();
         assertThat(test.getHref()).isNotBlank();
     }
+
+    @Test
+    void send_getSenderPhoneNumberFromConfigurationMap() {
+        CommunicationSpec.Twilio cfg = new CommunicationSpec.Twilio();
+        cfg.setKey("senderKey");
+        cfg.setAccountSid("AC528fe950968998cae3d3df2ac4f64fc0");
+        cfg.setAuthToken("5c93979e622a013ce781bb9b51eba6b8");
```
Comment (sergeysenja1992) [inline src/test/java/com/icthh/xm/tmf/ms/communication/service/TwilioServiceTest.java:73]: I hope it`s not real token

Comment (osiievska23) [reply src/test/java/com/icthh/xm/tmf/ms/communication/service/TwilioServiceTest.java:73]: they already been there when i come :)

Comment (osiievska23) [reply src/test/java/com/icthh/xm/tmf/ms/communication/service/TwilioServiceTest.java:73]: see beforeEach()

File: src/main/java/com/icthh/xm/tmf/ms/communication/messaging/template/AbstractMessageTemplateService.java (hunk 1 above)
Comment (osiievska23) [reply src/main/java/com/icthh/xm/tmf/ms/communication/messaging/template/AbstractMessageTemplateService.java]: fixed

File: src/test/java/com/icthh/xm/tmf/ms/communication/messaging/template/TwilioMessageTemplateServiceUnitTest.java (hunk 2 above)
Comment (osiievska23) [reply src/test/java/com/icthh/xm/tmf/ms/communication/messaging/template/TwilioMessageTemplateServiceUnitTest.java]: fixed

File: src/main/java/com/icthh/xm/tmf/ms/communication/messaging/template/AbstractMessageTemplateService.java [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+import java.util.Map;
+
+@Slf4j
+public abstract class AbstractMessageTemplateService implements MessageTemplateService {
+
+    private static final String TEMPLATE_NAME_MESSAGE_CHARACTERISTIC = "templateName";
+    private static final String LANGUAGE_MESSAGE_CHARACTERISTIC = "language";
+    private static final String TEMPLATE_MODEL_MESSAGE_CHARACTERISTIC = "templateModel";
+
+    private final freemarker.template.Configuration freeMarkerConfiguration;
```
[the remark this answers is not in the document]
Comment (osiievska23) [reply src/main/java/com/icthh/xm/tmf/ms/communication/messaging/template/AbstractMessageTemplateService.java]: fixed

**Review by amedvedchuk — APPROVED** [review]

**Review by sergeysenja1992 — APPROVED** [review]

**Pull request merged**